src/main/java/com/checkmarx/jenkins/tools/CheckmarxInstaller.java

```diff
@@ -195,6 +237,44 @@ public Void call() throws IOException {
             return null;
         }
 
+        public static void copyURLToFile(URL source, String proxyStr, String proxyType, File destination, int connectionTimeoutMillis, int readTimeoutMillis) throws IOException {
+            URLConnection connection;
+            if (proxyStr == null) {
+                connection = source.openConnection();
+            } else {
+                try {
+                    URI proxyUrl = new URI(proxyStr);
+                    Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyUrl.getHost(), proxyUrl.getPort()));
+                    connection = source.openConnection(proxy);
+                    if (StringUtils.isNotEmpty(proxyUrl.getUserInfo())) {
+                        // Proxy With UserInfo Not Checked !
+                        String authHeader = new String(Base64.getEncoder().encode(proxyUrl.getUserInfo().getBytes(UTF_8)), UTF_8).replace("\r\n", "");
+                        connection.setRequestProperty("Proxy-Authorization", getAuthProxyPrefix(proxyType) + authHeader);
+                        String[] userPass = proxyUrl.getUserInfo().split(":");
```

Comment:
split will throw if not exists : ?

Reply:
no, split will not throw an error.
but if ":" does not exist- the next line will fail - but it is OK to fail if the user enters a proxy with a wrong format.